overlays/nodejs/lambda/lambdaMixins.ts

```diff
@@ -0,0 +1,308 @@
+    readonly logGroupName: string;
+    readonly logStreamName: string;
+    readonly identity: any;
+    readonly clientContext: any;
+    getRemainingTimeInMillis(): string;
+}
+
+/**
+ * EntryPoint is the signature for a serverless function that will be invoked each time the AWS
+ * Lambda is invoked.
+ *
+ * It can be a synchronous or asynchronous function that can be converted into an AWS lambda.  Async
+ * callbacks are only supported with an AWS lambda runtime of 8.10 or higher.  On those runtimes a
+ * Promise can be returned, 'callback' parameter can be ignored, and AWS will appropriately handle
+ * things. For AWS lambda pre-8.10, a synchronous function must be provided.  The synchronous
+ * function should return nothing, and should instead invoke 'callback' when complete.
+ */
+export type EntryPoint<E, R> = (event: E, context: Context, callback: (error: any, result: R) => void) => Promise<R> | void;
+
+/**
+ * EntryPointFactory is the signature for a function that will be called once to produce the
+ * serverless function that AWS Lambda will invoke.  It can be used to initialize expensive state
+ * once that can then be used across all invocations of the Lambda (as long as the Lambda is using
+ * the same warm node instance).
+ */
+export type EntryPointFactory<E, R> = () => EntryPoint<E, R>;
+
+/**
+ * FunctionOptions provides configuration options for the serverless Function.  It is effectively
+ * equivalent to [aws.lambda.FunctionArgs] except with a few important differences documented at the
+ * property level.  For example, [role] is an actual iam.Role instance, and not an ARN. Properties
+ * like [runtime] are now optional.  And some properties (like [code]) are entirely disallowed.
+ */
+export type FunctionOptions<E, R> = utils.Overwrite<lambdaFunction.FunctionArgs, {
+    /**
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
+     */
+    code?: never;
+
+    /**
+     * Not allowed when creating an aws.serverless.Function.  The [code] will be generated from the
+     * passed in JavaScript callback.
+     */
+    handler?: never;
+
+    /**
+     * The Javascript function instance to use as the entrypoint for the AWS Lambda out of.  Either
+     * [func] or [factoryFunc] must be provided.
+     */
+    func?: EntryPoint<E, R>;
+
+    /**
+     * The Javascript function instance that will be called to produce the function that is the
+     * entrypoint for the AWS Lambda. Either [func] or [factoryFunc] must be provided.
+     *
+     * This form is useful when there is expensive initialization work that should only be executed
+     * once.  The factory-function will be invoked once when the final AWS Lambda module is loaded.
+     * It can run whatever code it needs, and will end by returning the actual function that Lambda
+     * will call into each time the Lambda is invoked.
+     */
+    factoryFunc?: EntryPointFactory<E, R>;
```

Comment:
Note: i would like to rename these properties o `entryPoint` and `entryPointFactory`.  however, with backcompat needs, that would mean still looking for these properties as well.  it seemed like too much effort, and 'func/factoryFunc' are acceptable.